lib/HLSL/DxilPreparePasses.cpp

```diff
@@ -640,6 +640,83 @@ class DxilFinalizeModule : public ModulePass {
     }
   }
 
+  void convertIsInf(Module &M, CallInst *CI) {
+    IRBuilder<> Builder(CI);
+    Value *Val = CI->getOperand(1);
+    Type *IType = Type::getInt16Ty(M.getContext());
+    Constant *PosInf = ConstantInt::get(IType, 0x7c00);
+    Constant *NegInf = ConstantInt::get(IType, 0xfc00);
+
+    Value *IVal = Builder.CreateBitCast(Val, IType);
+    Value *B1 = Builder.CreateICmpEQ(IVal, PosInf);
+    Value *B2 = Builder.CreateICmpEQ(IVal, NegInf);
+    Value *B3 = Builder.CreateOr(B1, B2);
+    CI->replaceAllUsesWith(B3);
+    CI->eraseFromParent();
+  }
+
+  void convertIsNaN(Module &M, CallInst *CI) {
+    IRBuilder<> Builder(CI);
+    Value *Val = CI->getOperand(1);
+    Type *IType = Type::getInt16Ty(M.getContext());
+
+    Constant *ExpBitMask = ConstantInt::get(IType, 0x7c00);
+    Constant *SigBitMask = ConstantInt::get(IType, 0x3ff);
+
+    Value *IVal = Builder.CreateBitCast(Val, IType);
+    Value *Exp = Builder.CreateAnd(IVal, ExpBitMask);
+    Value *B1 = Builder.CreateICmpEQ(Exp, ExpBitMask);
+
+    Value *Sig = Builder.CreateAnd(IVal, SigBitMask);
+    Value *B2 = Builder.CreateICmpNE(Sig, ConstantInt::get(IType, 0));
+    Value *B3 = Builder.CreateAnd(B1, B2);
+    CI->replaceAllUsesWith(B3);
+    CI->eraseFromParent();
+  }
+
+  void convertIsFinite(Module &M, CallInst *CI) {
+    IRBuilder<> Builder(CI);
+    Value *Val = CI->getOperand(1);
+    Type *IType = Type::getInt16Ty(M.getContext());
+
+    Constant *ExpBitMask = ConstantInt::get(IType, 0x7c00);
+
+    Value *IVal = Builder.CreateBitCast(Val, IType);
+    Value *Exp = Builder.CreateAnd(IVal, ExpBitMask);
+    Value *B1 = Builder.CreateICmpNE(Exp, ExpBitMask);
+    CI->replaceAllUsesWith(B1);
+    CI->eraseFromParent();
+  }
+
+  void convertIsSpecialFloat(Module &M, hlsl::OP *hlslOP) {
+    for (auto FnIt : hlslOP->GetOpFuncList(DXIL::OpCode::IsInf)) {
```

Comment:
I'm not convinced `FnIt` is a great name here.  It follows a similar convention used in this file, but I don't think the thing stored in this value is actually an iterator.

It's a...function overload maybe?  I did some digging and couldn't really understand what the two values stored in the "OpFuncList" actually are.  The second is a function.....the first..?